src/render/draft/draft_renderer.cpp

```diff
@@ -0,0 +1,166 @@
+#include "draft_renderer.h"
+
+cr::draft_renderer::draft_renderer(
+  uint64_t                    res_x,
+  uint64_t                    res_y,
+  std::unique_ptr<cr::scene> *scene)
+    : _res_x(res_x), _res_y(res_y), _scene(scene)
+{
+    glGenFramebuffers(1, &_framebuffer);
+    glBindFramebuffer(GL_FRAMEBUFFER, _framebuffer);
+
+    glGenTextures(1, &_texture);
+
+    glBindTexture(GL_TEXTURE_2D, _texture);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, _res_x, _res_y, 0, GL_RGB, GL_UNSIGNED_BYTE, nullptr);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+
+    glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, _texture, 0);
+
+    glGenRenderbuffers(1, &_rbo);
+    glBindRenderbuffer(GL_RENDERBUFFER, _rbo);
+    glRenderbufferStorage(GL_RENDERBUFFER, GL_DEPTH24_STENCIL8, _res_x, _res_y);
+
+    glFramebufferRenderbuffer(
+      GL_FRAMEBUFFER,
+      GL_DEPTH_STENCIL_ATTACHMENT,
+      GL_RENDERBUFFER,
+      _rbo);
+
+    if (glCheckFramebufferStatus(GL_FRAMEBUFFER) != GL_FRAMEBUFFER_COMPLETE)
+        fmt::print("[ERROR] Framebuffer is not complete");
+
+    glBindFramebuffer(GL_FRAMEBUFFER, 0);
+    glBindTexture(GL_TEXTURE_2D, 0);
+    glBindRenderbuffer(GL_FRAMEBUFFER, 0);
+
+    // Load shaders in
+    // Load the shader into the string
+    {
+        auto shader_file_in_stream = std::ifstream("./assets/shaders/shader.vert");
+        auto shader_string_stream  = std::stringstream();
+        shader_string_stream << shader_file_in_stream.rdbuf();
+        const auto shader_source = shader_string_stream.str();
+
+        // Create OpenGL shader
+        auto       shader_handle = glCreateShader(GL_VERTEX_SHADER);
+        const auto shader_string = shader_source.c_str();
+        glShaderSource(shader_handle, 1, &shader_string, nullptr);
+        glCompileShader(shader_handle);
+
+        auto success = int(0);
+        auto log     = std::array<char, 512>();
+        glGetShaderiv(shader_handle, GL_COMPILE_STATUS, &success);
+
+        if (!success)
+        {
+            glGetShaderInfoLog(shader_handle, 512, nullptr, log.data());
+            fmt::print("[ERROR] Compiling shader [{}], with error [{}]\n", "vertex", log.data());
```

Comment:
Another error that might want to be fatal, since the shader is constant. This might mean a driver bug.